app/src/components/UnitForm/index.js

```diff
@@ -1,188 +1,353 @@
+                    name="pet_friendly"
+                    checked={checked}
+                    onClick={handleCheck}
+                  />
+                }
+                label="Pet Friendly"
+              />
+              <TextField
+                name="laundry_availability"
+                variant="outlined"
+                label="Laundry Availability"
+                select
+                required
+                value={newbuilding.laundry_availability}
+                onChange={handleBuilding}
+              >
+                {laundryOptions.map((option) => (
+                  <MenuItem value={option}>{option}</MenuItem>
+                ))}
+              </TextField>
+              <TextField
+                name="type_of_unit"
+                variant="outlined"
+                label="Type of Unit"
+                select
+                required
+                value={newbuilding.type_of_unit}
+                onChange={handleBuilding}
+              >
+                {typeOptions.map((option) => (
+                  <MenuItem value={option}>{option}</MenuItem>
+                ))}
+              </TextField>
+              <TextField
+                name="distance_from_waterloo"
+                variant="outlined"
+                label="Distance From Waterloo"
+                required
+                value={newbuilding.distance_from_waterloo}
+                onChange={handleBuilding}
+                InputProps={{
+                  endAdornment: (
+                    <InputAdornment position="end">km</InputAdornment>
+                  ),
+                }}
+              />
+            </>
+          )}
+          <TextField
+            name="rent_price"
+            variant="outlined"
+            label="Price"
+            type="number"
+            sx={{ width: "33%" }}
+            required
+            value={postData.rent_price}
+            onChange={handleNum}
+          />
+          <TextField
+            name="num_beds"
+            variant="outlined"
+            label="Bedrooms"
+            type="number"
+            sx={{ width: "33%" }}
+            required
+            value={postData.num_beds}
+            onChange={handleNum}
+          />
+          <TextField
+            name="num_washrooms"
+            variant="outlined"
+            label="Washrooms"
+            type="number"
+            sx={{ width: "33%" }}
+            required
+            value={postData.num_washrooms}
+            onChange={handleNum}
           />
+          <TextField
+            name="floor_num"
+            variant="outlined"
+            label="Floor number"
+            type="number"
+            value={postData.floor_num}
+            sx={{ width: "33%" }}
+            onChange={handleNum}
+          />
+          <TextField
+            name="room_num"
+            variant="outlined"
+            label="Room Number"
+            type="number"
+            value={postData.room_num}
```

Comment:
Currently, I can submit a new building, not add any room/floor number, and the request will go through. We should either error out the request on the frontend, or just not allow the request to proceed from the frontend until all the data/state exists.

Reply:
Jenny suggested that room number or floor number are not required fields, so the user does not have to fill them in.

Reply:
@raveen-singh according to db schema they aren't requried fields